### PR TITLE
hw14

### PR DESCRIPTION
**Pull request by u2107680007911543774e7r**
Ready...
I have full files as delta : ))

### CODE REVIEW

File: src/main/java/lesson14_rest/MateGroupServiceImpl.java [hunk 1]
```diff
@@ -0,0 +1,153 @@
...
+			return Response.status(Status.ACCEPTED).build();
+		}
+		return Response.status(Status.NOT_FOUND).build();
+	}
+
+	@Override
+	@PUT
+	@Path("/hrs")
+	@Consumes(MediaType.APPLICATION_JSON)
+	@Produces(MediaType.APPLICATION_JSON)
```
Comment (u2107680007911543774e7r) [inline src/main/java/lesson14_rest/MateGroupServiceImpl.java:85]: I use Produces everywhere so I can make a single screenshot in postman : )

File: src/main/java/lesson14_rest/MateGroupServiceImpl.java [hunk 2]
```diff
@@ -0,0 +1,153 @@
...
+	@PUT
+	@Path("/hrs/{surname}")
+	@Produces(MediaType.APPLICATION_JSON)
+	public Response updateStartWorkYearHR(@PathParam("groupId") int groupId, @PathParam("surname") String surname,
+			@QueryParam("newStartWorkYear") int newStartWorkYear) {
+		MateGroup mateGroup = mateGroups.get(groupId);
+		if (mateGroup != null) {
+			mateGroup.getHumanResources().stream().filter((f -> f.getSurname().equals(surname)))
+					.collect(Collectors.toList()).forEach(f -> f.setStartWorkYear(newStartWorkYear));
+			return Response.status(Status.OK).entity(mateGroups.get(groupId).getHumanResources())
```
Comment (DadizhaO) [inline src/main/java/lesson14_rest/MateGroupServiceImpl.java:120]: collect(Collectors.toList()) it's unnecessary operation.

File: src/main/java/lesson14_rest/MateGroupServiceImpl.java (hunk 1 above)
Comment (DadizhaO) [reply src/main/java/lesson14_rest/MateGroupServiceImpl.java:85]: To use Produces not necessary for this it's works and without Produces and Consumes (and single screenshot too).

**Review by DadizhaO — APPROVED** [review]

File: src/main/java/lesson14_rest/MateGroupServiceImpl.java [hunk 3]
```diff
@@ -0,0 +1,153 @@
...
+		if (mateGroup != null) {
+			mateGroup.getStudents().stream().filter((f -> f.getSurname().equals(surname))).collect(Collectors.toList())
+					.forEach(f -> f.setName(newname));
+			return Response.status(Status.ACCEPTED).build();
+		}
+		return Response.status(Status.NOT_FOUND).build();
+	}
+
+	@Override
+	@PUT
```
Comment (SerhiiPasko) [inline src/main/java/lesson14_rest/MateGroupServiceImpl.java:82]: Post better there

**Review by SerhiiPasko — CHANGES_REQUESTED** [review]
Doing changes to show me only diff

**Review by SerhiiPasko — APPROVED** [review]

**Pull request merged**